src/renderer/src/hooks/useGetFullTransactions.tsx

```diff
@@ -0,0 +1,333 @@
+    }),
+    queryFn: ({ pageParam: page }) =>
+      getFullTransactions({
+        queryClient,
+        dateFrom,
+        dateTo,
+        accounts,
+        allAccounts,
+        networksByBlockchain,
+        page,
+      }),
+    initialPageParam: 1,
+    getNextPageParam: ({ page }) => page,
+  })
+
+  const data = useMemo(() => {
+    if (query.isLoading) return []
+
+    let groupedData = new Map<string, TFullTransactionsItem>()
+    const data = cloneDeep(query.data?.pages?.flatMap(page => page?.data ?? []) ?? [])
+
+    data.forEach(items => {
+      groupedData = new Map<string, TFullTransactionsItem>([...groupedData, ...items])
+    })
+
+    const insertedPendingTransferHashes = new Set<string>()
+
+    pendingTransactions
+      .toSorted(({ time: newerTime }, { time: olderTime }) => newerTime - olderTime)
+      .forEach(({ hash, time, account, to, from, assetHash, token, ...transfer }) => {
+        if (
+          !accounts.some(AccountHelper.predicate(account)) ||
+          !isWithinInterval(createNewDateByTime(time), { start: dateFrom, end: dateTo })
+        )
+          return
+
+        const { blockchain } = account
+        const service = bsAggregator.blockchainServicesByName[blockchain]
+
+        let txTemplateUrl: string | undefined
+        let addressTemplateUrl: string | undefined
+        let contractTemplateUrl: string | undefined
+
+        if (hasExplorerService(service)) {
+          txTemplateUrl = service.explorerService.getTxTemplateUrl()
+          addressTemplateUrl = service.explorerService.getAddressTemplateUrl()
+          contractTemplateUrl = service.explorerService.getContractTemplateUrl()
+        }
+
+        const event: TFullTransactionAssetEvent = {
+          hash: assetHash,
+          eventType: 'token',
+          amount: transfer.amount,
+          methodName: 'transfer',
+          hashUrl: contractTemplateUrl?.replace('{hash}', assetHash),
+          tokenType: 'generic',
+          token,
+          from,
+          fromUrl: addressTemplateUrl?.replace('{address}', from),
+          fromAccount: transfer.fromAccount,
+          to,
+          toUrl: addressTemplateUrl?.replace('{address}', to),
+          toAccount: transfer.toAccount,
+        }
+
+        const wasAlreadyInsertedHash = insertedPendingTransferHashes.has(hash)
+
+        if (wasAlreadyInsertedHash) {
+          const transaction = groupedData.get(hash)!
+
+          transaction.events.push(event)
+
+          return
+        }
+
+        groupedData.set(hash, {
+          txId: hash,
+          txIdUrl: txTemplateUrl?.replace('{txId}', hash),
+          block: 0,
+          date: createNewDateByTime(time).toISOString(),
+          invocationCount: 0,
+          notificationCount: 0,
+          networkFeeAmount: '0',
+          systemFeeAmount: '0',
+          blockchain,
+          isPending: true,
+          events: [event],
+        })
+
+        insertedPendingTransferHashes.add(hash)
+      })
```

Comment:
Instead of doing a toSorted and a forEach, you can do an inverted for each, since the last pending ones are always the new ones.

```
for (let index = pendingTransactions.lenght - 1; index >= 0; index--) {
 const item = pendingTransactions[index]
 ...rest 
}
```